core/src/main/scala/com/nvidia/spark/rapids/tool/analysis/AppSparkMetricsAnalyzer.scala

```diff
@@ -438,7 +438,7 @@ class AppSparkMetricsAnalyzer(app: AppBase) extends AppAnalysisBase(app) {
         val peakMemoryValues = tasksInStage.map(_.peakExecutionMemory)
         val shuffleWriteTime = tasksInStage.map(_.sw_writeTime)
         (AppSparkMetricsAnalyzer.maxWithEmptyHandling(peakMemoryValues),
-          shuffleWriteTime.sum)
+          TimeUnit.NANOSECONDS.toMillis(shuffleWriteTime.sum))
```

Comment:
This still does not fix the problem because the conversion is done on the stage-level.
The correct way, is to convert after the metrics are aggregated on each level.
For example, perStage/perSql/perJob.

Reply:
Correct!
But when we are aggregating perSql, this PR is actually aggregating the stages per SQL after the time is converted to milliseconds.
If we want to be more accurate, then the cached-per-stage-results should still be in nano-seconds; then per-sql value is the sum in nano-seconds; and finally it gets converted to milliseonds.

Reply:
understood, thanks @amahussein! I will address this now.

Reply:
Discussed offline. We will keep the current implementation to avoid potential overflow if we aggregate at SQL/job level.